File: src/test/java/com/github/bartgora/GameCharacterTest.java
package com.github.bartgora;


import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.*;

class GameCharacterTest {

    @Test
    void shouldDisplayGandalf(){
        //given
        GameCharacter characterTemplate = new WizardCharacter();
        //then
        var rendered = characterTemplate.render();
        assertThat(rendered).isEqualTo("My name is Gandalf I have a Stick And I can Fly");
    }
}

File: src/test/kotlin/com/github/bartgora/GameCharacterTest.kt
package com.github.bartgora

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class GameCharacterTest {

    @Test
    fun shouldDisplayGandalf() {
        //given
        val characterTemplate: GameCharacter = WizardCharacter()
        //then
        val rendered = characterTemplate.render()
        rendered shouldBe "My name is Gandalf I have a Stick And I can Fly"
    }
}